rfcs/0024-Non-consensus-node-protocol.md

```diff
@@ -0,0 +1,219 @@
+
+Now that NC nodes have the merkle root of the latest ledger, they can request merkle paths to their accounts from peers that store the full state and verify it agaisnt the staged ledger hash. With the account state from peers, an NC node can:
+
+1. Check the balance
+2. Get the latest account nonce and use it to send transactions.(Also, infer the latest nonce from local transaction pool)
+3. Check the status of sent transactions.This is achieved by comparing the receipt-chain hash from the account state.
+For example if an NC node sends transactions t1, t2, and t3 in that order and the reciept chain hashes are r1, r2, and r3 and if the account receipt-chain hash is r2 it would mean that t1 and t2 were included in the block and t3 is still pending.
+
+An NC node can delegate an account's state using the existing command `coda client delegate-stake`
+
+##### Block header at some block k (root of the transition frontier)
+
+For wallets to be able to show if transactions are finalized, non-consensus nodes will need to track the state of block at some length `k` such that the probablity of a fork of length `k` is extremely low.
+
+On startup, a non-consensus node will request from its peers the block header at root and subsequent changes in the root need to be gossiped (layer 2). Given the root and the best tip, NC nodes should be able to verify that they are indeed on the same chain. This is currently done using a merkle list of state hashes between the root and the best tip and the same can be used for non-consensus nodes as well.
+
+With the block header at the root, NC nodes can query the account state from the staged ledger materialized from the snarked ledger.
+
+The current value of k is 2160 which means for a transaction to be finalized, it would take 10 days. That does not seem like a good user experience. Transactions on Bitcoin and Ethereum are finalized after 6 (~an hour) and 30 (~7mins) block respectively.
```

Comment:
Bitcoin level of finality is much much smaller than 2160. I think we can consider k for the purposes of ouroboros distinct from the k blocks back a user would want to wait to trust finality